### ISSUE TITLE
TST/DOC: updates to `list_files` methods for test instruments

### ISSUE THREAD

**Issue by jklenzing** [pysat/pysat#631]
**Is your feature request related to a problem? Please describe.**
It is currently not clear that test instruments can only "load data" for 2008-2010.  This should be documented and perhaps extended for use in packages that rely on pysat.

**Describe the solution you'd like**
- [x] Documentation should be updated to reflect usage
- [x] Extension / modification of usable dates?

**Describe alternatives you've considered**

**Additional context**
related to pysat/pysatMissions#42

**Comment by jklenzing** [comment]
Possibility: add keyword so that `Files` auto-generate filenames

**Comment by jklenzing** [comment]
3.0.0-related milestones fixed in #649.

**Comment by rstoneback** [comment]
Extended keyword arguments to list_files and associated areas so `file_date_range` can be provided at instantiation or later in #727 across all test instruments.
```
In [8]: test = pysat.Instrument('pysat', 'testing', file_date_range=pds.date_range(dt.datetime(2020, 1, 1), dt.datetime(2021, 1, 1)))                                                                                       

In [9]: test.files.files                                                                                                                                                                                                    
Out[9]: 
2020-01-01    2020-01-01.nofile
2020-01-02    2020-01-02.nofile
2020-01-03    2020-01-03.nofile
2020-01-04    2020-01-04.nofile
2020-01-05    2020-01-05.nofile
                    ...        
2020-12-28    2020-12-28.nofile
2020-12-29    2020-12-29.nofile
2020-12-30    2020-12-30.nofile
2020-12-31    2020-12-31.nofile
2021-01-01    2021-01-01.nofile
Freq: D, Length: 367, dtype: object
```

```
In [28]: In [23]: test.kwargs['list_files']['file_date_range'] = pds.date_range(dt.datetime(2021, 1, 1), dt.datetime(2022, 1, 1))                                                                                           
    ...:           
    ...:  
    ...: In [24]: test.files.refresh()                                                                                                                                                                                      
    ...:           
    ...:  
    ...: In [25]: test.files.files                                                                                                                                                                                          
    ...:           
    ...:                                                                                                                                                                                                                    
Out[28]: 
2021-01-01    2021-01-01.nofile
2021-01-02    2021-01-02.nofile
2021-01-03    2021-01-03.nofile
2021-01-04    2021-01-04.nofile
2021-01-05    2021-01-05.nofile
                    ...        
2021-12-28    2021-12-28.nofile
2021-12-29    2021-12-29.nofile
2021-12-30    2021-12-30.nofile
2021-12-31    2021-12-31.nofile
2022-01-01    2022-01-01.nofile
Freq: D, Length: 366, dtype: object
```

**Comment by jklenzing** [comment]
Solved long ago, integration tests added in #878
